### PR TITLE
Added the O3 shortcut

### PR DESCRIPTION
**Pull request by PatrickWaweru**
Added the O3 shortcut

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.openmrs.ui.framework.page.PageModel;
+import org.openmrs.util.OpenmrsUtil;
+
+@AppPage(EmrConstants.APP_O3_SHORTCUT)
+public class O3HomePageController {
+
+    public void get(@SpringBean KenyaUiUtils kenyaUi, UiUtils ui, PageModel model) {
+        System.err.println("o3 shortcut Page get is hit : " + model.getAttribute("currentPatient"));
+        // Get apps for the current user
+		List<AppDescriptor> apps = Context.getService(AppFrameworkService.class).getAppsForCurrentUser();
```
Comment (ojwanganto) [inline omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java:37]: what do we use this for?

File: omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			ret.put("patientId", patient.getId());
+			ret.put("patientUUID", patient.getUuid());
+		} else {
+			ret.put("patientExists", false);
+			ret.put("patientId", "");
+			ret.put("patientUUID", "");
+		}
+
+        model.addAttribute("patientDetails", ui.toJson(ret));
+		model.addAttribute("apps", apps);
```
Comment (ojwanganto) [inline omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java:60]: do we need this on the view?

**Review by ojwanganto — CHANGES_REQUESTED** [review]

File: api/src/main/resources/moduleApplicationContext.xml [hunk 3]
```diff
@@ -142,6 +142,14 @@
 		<constructor-arg value="840" />
 	</bean>
 
+	<bean id="kenyaemr.app.openmrs3Shortcut" factory-bean="kenyacore.appFactoryBean" factory-method="createInstance">
+		<constructor-arg value="kenyaemr.openmrs3Shortcut" />
+		<constructor-arg value="KenyaEMR 3.x" />
```
[the remark this answers is not in the document]
Comment (PatrickWaweru) [reply api/src/main/resources/moduleApplicationContext.xml:147]: Yes that looks ok

File: omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java (hunk 1 above)
Comment (PatrickWaweru) [reply omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java:37]: Yes we don't need it. Let me remove

File: omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java (hunk 2 above)
Comment (PatrickWaweru) [reply omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java:60]: Yes we don't need it. Let me remove

Comment (PatrickWaweru) [reply omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java]: Fixed on https://github.com/palladiumkenya/openmrs-module-kenyaemr/pull/1393/commits/4ca590cafe33ea177864af6badfffc38f2a8c39a

File: omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java (hunk 1 above)
Comment (PatrickWaweru) [reply omod/src/main/java/org/openmrs/module/kenyaemr/page/controller/shortcuts/O3HomePageController.java]: Fixed on https://github.com/palladiumkenya/openmrs-module-kenyaemr/pull/1393/commits/4ca590cafe33ea177864af6badfffc38f2a8c39a

File: api/src/main/resources/moduleApplicationContext.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (PatrickWaweru) [reply api/src/main/resources/moduleApplicationContext.xml]: Fixed on https://github.com/palladiumkenya/openmrs-module-kenyaemr/pull/1393/commits/4ca590cafe33ea177864af6badfffc38f2a8c39a

**Review by ojwanganto — APPROVED** [review]

**Pull request merged**